src/test/java/com/commercetools/sync/producttypes/utils/ProductTypesUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,99 @@
+package com.commercetools.sync.producttypes.utils;
+
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.models.LocalizedString;
+import io.sphere.sdk.products.attributes.AttributeDefinition;
+import io.sphere.sdk.products.attributes.AttributeDefinitionBuilder;
+import io.sphere.sdk.products.attributes.AttributeDefinitionDraft;
+import io.sphere.sdk.products.attributes.AttributeDefinitionDraftBuilder;
+import io.sphere.sdk.products.attributes.StringAttributeType;
+import io.sphere.sdk.producttypes.ProductType;
+
+import static com.commercetools.sync.producttypes.utils.ProductTypeUpdateActionUtils.buildChangeDescriptionAction;
+import static com.commercetools.sync.producttypes.utils.ProductTypeUpdateActionUtils.buildChangeNameAction;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.sphere.sdk.producttypes.ProductTypeDraft;
+import io.sphere.sdk.producttypes.commands.updateactions.ChangeDescription;
+import io.sphere.sdk.producttypes.commands.updateactions.ChangeName;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+
+public class ProductTypesUpdateActionUtilsTest {
+    private static ProductType old;
+    private static ProductTypeDraft newSame;
+    private static ProductTypeDraft newDifferent;
+
+    /**
+     * Initialises test data.
+     */
+    @BeforeClass
+    public static void setup() {
+        final LocalizedString label = LocalizedString.of(Locale.ENGLISH, "label1");
+        final AttributeDefinition attributeDefinition = AttributeDefinitionBuilder
+                .of("attributeName1", label, StringAttributeType.of())
+                .build();
+
+        final List<AttributeDefinition> oldAttributeDefinitions = singletonList(attributeDefinition);
+
+        final List<AttributeDefinitionDraft> sameAttributeDefinitionDrafts = singletonList(
+                AttributeDefinitionDraftBuilder.of(attributeDefinition).build());
+
+        old = mock(ProductType.class);
+        when(old.getKey()).thenReturn("key1");
+        when(old.getName()).thenReturn("name1");
+        when(old.getDescription()).thenReturn("description 1");
+        when(old.getAttributes()).thenReturn(oldAttributeDefinitions);
+
+        newSame = ProductTypeDraft.ofAttributeDefinitionDrafts(
+                "key1",
+                "name1",
+                "description 1",
+                sameAttributeDefinitionDrafts
+        );
+
+        newDifferent = ProductTypeDraft.ofAttributeDefinitionDrafts(
+                "key2",
+                "name2",
+                "description 2",
+                sameAttributeDefinitionDrafts
```

Comment:
I know you don't test attribute definitions but why use the same defintions in the `newDifferent` ProductTypeDraft?

Is this something that you will change later? If so, maybe add a `TODO` so it's not looked over?

Reply:
The attributes definitions are required so in order to provide a value I use the same. 

I think using the same or different attributes definitions doesn't make any difference right now because the unit tests at this stage are focused at the product type update actions levels (name, description)

The tests for attribute definitions update actions are included in a separate file that is not included in this PR.

I add a TODO to clarify that the attribute definitions will change later